packages/deepsignal/core/test/types.ts

```diff
@@ -74,3 +75,69 @@ store2.a = a;
 store2.$a = a;
 const s5: number = store2.a!;
 const s6: Signal<number | undefined> = store2.$a;
+
+// Shallow
+const store3 = deepSignal({
+	a: { b: 1 },
+	c: shallow({ b: 2 }),
+});
+
+store3.a.$b;
+// @ts-expect-error
+store3.c.$b;
+
+store3.a = { b: 1 };
+// @ts-expect-error
+store3.c = { b: 2 };
+store3.c = shallow({ b: 2 });
+
+type Store4 = {
+	a: { b: number };
+	c: Shallow<{ b: number }>;
+	d: Shallow<{ b: number }>;
+};
+
+const store4 = deepSignal<Store4>({
+	a: { b: 1 },
+	c: shallow({ b: 2 }),
+	// @ts-expect-error
+	d: { b: 3 },
+});
+
+store4.a.$b;
+// @ts-expect-error
+store4.c.$b;
+
+// Manual typings
+type Store5 = {
+	a: { b: number };
+	c: { b: number };
+};
+const store5 = deepSignal<Store5>({
+	a: { b: 1 },
+	c: { b: 1 },
+});
+
+store5.a.b;
+store5.a.$b;
+store5.c.b;
+store5.c.$b;
+// @ts-expect-error
+store5.d;
+
+type Store6 = {
+	[key: string]: { b: number };
+};
+
+const store6 = deepSignal<Store5>({
```

Comment:
Maybe you wanted to use `Store6` type rather than `Store5`? I see `Store6` is not used anywhere. 🤔

Reply:
Yes, thanks. Fixed in https://github.com/luisherranz/deepsignal/pull/38/commits/2e24d23734934de7695dfa727af9647a4d7a80c1.